tabular/src/autogluon/tabular/task/tabular_prediction/dataset.py

```diff
@@ -69,36 +49,11 @@ def _constructor(self):
     def _constructor_sliced(self):
         return pd.Series
 
-    def __init__(self, *args, **kwargs):
-        file_path = kwargs.get('file_path', None)
-        name = kwargs.get('name', None)
-        feature_types = kwargs.get('feature_types', None)
-        df = kwargs.get('df', None)
-        subsample = kwargs.get('subsample', None)
-        copy = kwargs.get('copy', False)
-        construct_from_df = False  # whether or not we are constructing new dataset object from scratch based on provided DataFrame.
-        # if df is None and file_path is None: # Cannot be used currently!
-        #     raise ValueError("Must specify either named argument 'file_path' or 'df' in order to construct tabular Dataset")
-        if df is not None:  # Create Dataset from existing Python DataFrame:
-            construct_from_df = True
-            if not isinstance(df, pd.DataFrame):
-                raise ValueError("'df' must be existing pandas DataFrame. To read dataset from file instead, use 'file_path' string argument.")
-            if file_path is not None:
-                warnings.warn("Both 'df' and 'file_path' supplied. Creating dataset based on DataFrame 'df' rather than reading from file_path.")
-            if copy:
-                df = df.copy(deep=True)
-        elif file_path is not None:  # Read from file to create dataset
-            construct_from_df = True
-            df = load_pd.load(file_path)
-        if construct_from_df:  # Construct new Dataset object based off of DataFrame
-            if subsample is not None:
-                if not isinstance(subsample, int) or subsample <= 1:
-                    raise ValueError("'subsample' must be of type int and > 1")
-                df = df.head(subsample)
-            super().__init__(df)
-            self.file_path = file_path
-            self.name = name
-            self.feature_types = feature_types
-            self.subsample = subsample
+    def __init__(self, data, **kwargs):
+        if isinstance(data, str):
+            file_path = data
+            data = load_pd.load(file_path)
```

Comment:
consider accepting `pd.read_csv` kwargs passed through here into `load_pd.load`, so user can specify additional things (eg. tab vs comma delimiter)

Reply:
Unfortunately it isn't so simple since we actually use TabularDataset the same way as pd.DataFrame, and init is called internally by pandas ('data' is the first argument in pd.DataFrame()), so we can't add other parameters. If we wanted to separate the init of TabularDataset from the object itself, it would work (via .from_csv, .from_df, .from_parquet, etc.), but then someone would have to know which one to specify.

Reply:
Maybe at least documentation should more clearly state: the delimiter restrictions, and that user should use `pd.read_csv()` instead to read files with other delimiters?

Reply:
Can we add delimiter parameter then? Tab-separated files is a very frequent format to support. We can also infer it from extension

Reply:
I don't think we should add any extra parameters to init, since this is called many times by pandas (You can see for yourself if you add a print statement in the init, it is called many times). Instead, for more advanced file loading options, I think it makes sense to have a `TabularDataset.load_csv`, `TabularDataset.load_parquet`, etc. These wouldn't be accessed by pandas so we could do whatever we want with them.